benchmarks/apache-spark/src/main/scala/org/renaissance/apache/spark/ZipResourceUtil.scala

```diff
@@ -1,21 +1,40 @@
 package org.renaissance.apache.spark
 
+import java.io.FileNotFoundException
 import java.nio.charset.StandardCharsets
 import java.util.zip.ZipInputStream
 
-import org.apache.commons.io.IOUtils
+import scala.io.BufferedSource
+import scala.io.Source
 
 object ZipResourceUtil {
 
-  def readZipFromResourceToText(resourceName: String): String = {
-    val zis = new ZipInputStream(this.getClass.getResourceAsStream("/" + resourceName))
+  def sourceFromZipResource(entryName: String, resourceName: String): BufferedSource = {
+    val is = this.getClass.getResourceAsStream("/" + resourceName)
+    if (is == null) {
+      throw new FileNotFoundException(
+        s"resource '$resourceName' not found"
```

Comment:
Perhaps prepend `/` here as well? Or the caller should take care of specifying absolute path.

Reply:
You are right. I modified the utility method to require the user to pass the absolute path, it makes it more clear what is requested at the caller side.